subgraphs/osmosis/src/modules/Metrics.ts

```diff
@@ -0,0 +1,416 @@
+  poolHourlySnaphot.hourlyVolumeByTokenUSD = hourlyVolumeByTokenUSD;
+
+  poolHourlySnaphot.save();
+  poolDailySnaphot.save();
+}
+
+export function updateSnapshotsVolume(
+  liquidityPoolId: string,
+  volumeUSD: BigDecimal,
+  block: cosmos.HeaderOnlyBlock
+): void {
+  const protcol = getOrCreateDexAmmProtocol();
+  const financialsDailySnapshot = getOrCreateFinancialDailySnapshots(block);
+  const liquidityPool = LiquidityPoolStore.load(liquidityPoolId);
+  if (liquidityPool == null) {
+    return;
+  }
+  const poolDailySnaphot = getOrCreateLiquidityPoolDailySnapshots(
+    liquidityPoolId,
+    block
+  );
+  const poolHourlySnaphot = getOrCreateLiquidityPoolHourlySnapshots(
+    liquidityPoolId,
+    block
+  );
+  if (poolDailySnaphot == null || poolHourlySnaphot == null) {
+    return;
+  }
+
+  const blockNumber = BigInt.fromI32(block.header.height as i32);
+  const timestamp = BigInt.fromI32(block.header.time.seconds as i32);
+
+  financialsDailySnapshot.dailyVolumeUSD = financialsDailySnapshot.dailyVolumeUSD.plus(
+    volumeUSD
+  );
+  financialsDailySnapshot.cumulativeVolumeUSD = financialsDailySnapshot.cumulativeVolumeUSD.plus(
+    volumeUSD
+  );
+  financialsDailySnapshot.totalValueLockedUSD = financialsDailySnapshot.totalValueLockedUSD.plus(
+    volumeUSD
+  );
+  financialsDailySnapshot.blockNumber = blockNumber;
+  financialsDailySnapshot.timestamp = timestamp;
+
+  poolDailySnaphot.dailyVolumeUSD = poolDailySnaphot.dailyVolumeUSD.plus(
+    volumeUSD
+  );
+  poolDailySnaphot.cumulativeVolumeUSD = liquidityPool.cumulativeVolumeUSD;
+  poolDailySnaphot.totalValueLockedUSD = liquidityPool.totalValueLockedUSD;
+  poolDailySnaphot.inputTokenBalances = liquidityPool.inputTokenBalances;
+  poolDailySnaphot.inputTokenWeights = liquidityPool.inputTokenWeights;
+  poolDailySnaphot.outputTokenSupply = liquidityPool.outputTokenSupply;
+  poolDailySnaphot.outputTokenPriceUSD = liquidityPool.outputTokenPriceUSD;
+  poolDailySnaphot.blockNumber = blockNumber;
+  poolDailySnaphot.timestamp = timestamp;
+
+  poolHourlySnaphot.hourlyVolumeUSD = poolHourlySnaphot.hourlyVolumeUSD.plus(
+    volumeUSD
+  );
+  poolHourlySnaphot.cumulativeVolumeUSD = liquidityPool.cumulativeVolumeUSD;
+  poolHourlySnaphot.totalValueLockedUSD = liquidityPool.totalValueLockedUSD;
+  poolHourlySnaphot.inputTokenBalances = liquidityPool.inputTokenBalances;
+  poolHourlySnaphot.inputTokenWeights = liquidityPool.inputTokenWeights;
+  poolHourlySnaphot.outputTokenSupply = liquidityPool.outputTokenSupply;
+  poolHourlySnaphot.outputTokenPriceUSD = liquidityPool.outputTokenPriceUSD;
+  poolHourlySnaphot.blockNumber = blockNumber;
+  poolHourlySnaphot.timestamp = timestamp;
+
+  protcol.cumulativeVolumeUSD = protcol.cumulativeVolumeUSD.plus(volumeUSD);
+
+  financialsDailySnapshot.save();
+  poolDailySnaphot.save();
+  poolHourlySnaphot.save();
+  protcol.save();
+}
+
+export function updateProtocolRevenue(
```

Comment:
Is this even used? Why have updateProtocolRevenue and updateSupplyRevenue?

Reply:
Deleted updateProtocolRevenue() since it is not used at https://github.com/messari/subgraphs/commit/6903f52ce28319ef61b2eb05dafdabf73a47568c

Reply:
Does protocol revenue get updated anywhere then? Is there protocol revenue?

Reply:
Right now protocol revenue get updated at updateRevenueSnapshots(). There is not protocol revenue for osmosis dex yet.